src/WebJobs.Extensions.OpenAI.AISearch/AzureAISearchProvider.cs

```diff
@@ -0,0 +1,323 @@
+    {
+        int iteration = 0;
+        IndexDocumentsBatch<SearchDocument> batch = new();
+        for (int i = 0; i < document.Embeddings.Response.Data.Count; i++)
+        {
+            batch.Actions.Add(new IndexDocumentsAction<SearchDocument>(
+                IndexActionType.MergeOrUpload,
+                new SearchDocument
+                {
+                    ["id"] = Guid.NewGuid().ToString("N"),
+                    ["text"] = document.Embeddings.Request.Input![i],
+                    ["title"] = Path.GetFileNameWithoutExtension(document.Title),
+                    ["embeddings"] = document.Embeddings.Response.Data[i].Embedding.ToArray() ?? Array.Empty<float>(),
+                    ["timestamp"] = DateTime.UtcNow
+                }));
+            iteration++;
+            if (iteration % 1_000 is 0)
+            {
+                // Every one thousand documents, batch create.
+                IndexDocumentsResult result = await searchClient.IndexDocumentsAsync(batch, cancellationToken: cancellationToken);
+                int succeeded = result.Results.Count(r => r.Succeeded);
+                this.logger.LogDebug("""
+                        Indexed {Count} sections, {Succeeded} succeeded
+                        """,
+                        batch.Actions.Count,
+                        succeeded);
+
+                batch = new();
+            }
+        }
+
+        if (batch is { Actions.Count: > 0 })
+        {
+            // Any remaining documents, batch create.
+            IndexDocumentsResult result = await searchClient.IndexDocumentsAsync(batch, cancellationToken: cancellationToken);
+            int succeeded = result.Results.Count(r => r.Succeeded);
+            this.logger.LogDebug("""
+                        Indexed {Count} sections, {Succeeded} succeeded
+                        """,
+                    batch.Actions.Count,
+                    succeeded);
+        }
+    }
+
+    static SearchIndexClient GetSearchIndexClient(string endpoint, string key)
+    {
+        if (string.IsNullOrEmpty(key))
+        {
+            return new SearchIndexClient(new Uri(endpoint), new DefaultAzureCredential());
+        }
+        else
+        {
+            return new SearchIndexClient(new Uri(endpoint), new AzureKeyCredential(key));
+        }
+    }
+
+    static SearchClient GetSearchClient(string endpoint, string key, string searchIndexName)
+    {
+        SearchClient searchClient;
+        if (string.IsNullOrEmpty(key))
+        {
+            searchClient = new SearchClient(new Uri(endpoint), searchIndexName, new DefaultAzureCredential());
+        }
+        else
+        {
+            searchClient = new SearchClient(new Uri(endpoint), searchIndexName, new AzureKeyCredential(key));
+        }
+
+        return searchClient;
+    }
+
+    public ISearchProvider GetServiceProvider(string serviceProviderName)
+    {
+        throw new NotImplementedException();
```

Comment:
What's this method for? Can it be removed?

Reply:
it was residue of my trial to do run time configuration of Search Provider, removed it now.